### ISSUE TITLE
Getting incorrect page orientation value

### ISSUE THREAD

**Issue by vikasr111** [mindee/doctr#1801]
### Bug description

I am working on OCR scenarios where documents can be rotated. I have set `detect_orientation=True` to observe the page orientation value. I am noticing inconsistent result for page orientation. I have a couple of documents where one is rotated 90 degree left and other is rotated 90 degree to right.

For both the documents I am getting orientation value `{'value': 90, 'confidence': None}`. How can I get the correct value of rotation which I can use to rotate the page to the right orientation. My goal is to reconstruct or rotate the page to it's right orientation so that I can display the pdf properly.

Here are the sample documents:

![bol-r1_page-0001](https://github.com/user-attachments/assets/f7a881a8-0b29-45f2-80a7-fbae6d57286d)

![bol-r2_page-0001](https://github.com/user-attachments/assets/697df943-2093-48df-ab64-b093922e6350)

### Code snippet to reproduce the bug

```
from doctr.models import ocr_predictor
from doctr.io import DocumentFile

pdf_path = "some path"
docs = DocumentFile.from_pdf(pdf_path)

# load model
predictor = ocr_predictor(
    det_arch="db_mobilenet_v3_large",
    reco_arch="parseq",
    pretrained=True,
    preserve_aspect_ratio=False,
    symmetric_pad=False,
    )

predictor.det_predictor.model.postprocessor.bin_thresh = 0.35
predictor.det_predictor.model.postprocessor.box_thresh = 0.3


result = predictor(docs)
ocr_json = result.export()

for page in ocr_json.get("pages", []):
      page_data = {
          "page_idx": page.get("page_idx"),
          "dimensions": page.get("dimensions"),
          "orientation": page.get("orientation"),
          "lines": [],
      }
      print(f"Page orientation: {page.get('orientation')}")
```

### Error traceback

The page orientation value is incorrect.

### Environment

Python 3.11.7

### Deep Learning backend

Torch

**Comment by felixdittrich92** [comment]
Hi @vikasr111 :wave:,

You can easily fine tune the orientation models to your needs :)

See: https://mindee.github.io/doctr/using_doctr/custom_models_training.html#custom-orientation-classification-models

And https://github.com/mindee/doctr/tree/main/references/classification#usage-orientation-classification


Only images are required for training no annotations / all images for training shouldn't be rotated (+-10 degree would be fine) - labels are created by auto augmentation

Hope this helps :)

Best,
Felix

**Comment by vikasr111** [comment]
Thanks @felixdittrich92.
I handled this issue using a custom orientation fix using opencv. Would love to share the solution.

**Comment by felixdittrich92** [comment]
Hey @vikasr111 :wave:,

Sure would be great to share a snippet here where we can discuss if it makes sense to integrate in any way into docTR :hugs:

**Comment by aymen20002005** [comment]
Hello @vikasr111,
Can you share with us how you fixed it pls?
Thank you,

**Comment by vikasr111** [comment]
@felixdittrich92 @aymen20002005 Earlier I had written a whole module which uses pdf2image to extract page image and run DocTR to find the orientation of the image. Later I realized that it's not the problem of DocTR but OnnxTR which I was using earlier.

I want to report a strange case where for the same image DocTR gives correct orientation of 0 as the image is vertical, while OnnxTR gives 180degree orientation.

Here's the image:
![abc6_page_1](https://github.com/user-attachments/assets/8256c373-0e90-4c21-9772-4958a1208017)


Here's the code snippet:

```
import os
import json
from doctr.io import DocumentFile
from doctr.models import ocr_predictor
# from onnxtr.io import DocumentFile
# from onnxtr.models import ocr_predictor

os.environ['USE_TORCH'] = '1'


img_path = "extracted_image/abc6_page_1.png"


predictor = ocr_predictor(
    det_arch="db_resnet50",
    reco_arch="crnn_mobilenet_v3_large",
    resolve_blocks=True,
    resolve_lines=True,
    paragraph_break=0.035,
    assume_straight_pages=False,
    detect_orientation=True,
)

doc = DocumentFile.from_images(img_path)
result = predictor(doc)
text_data = result.render()
json_data = result.export()
print(json_data)

parsed_file = f"doctr-output.txt"
with open(parsed_file, "w") as f:
    f.write(text_data)
```

**Comment by felixdittrich92** [comment]
Hey @vikasr111 :wave:,

Thanks for reporting i will have a look - OnnxTR uses the docTR models exported to Onnx while exporting it's possible that it leads to a small numerical difference - we check this already in our CI with an absolute tolerance from 0.0001 - but it's still possible that (especially the cases if the image is exactly 0, 90, 180, 270 degrees rotated) it differs (i expect it works in ~95% as the pytorch model).

I'm not sure if i can do anything about this but i will have a look :)

So do you think we can close the ticket here and move it to OnnxTR ? :)

**Comment by vikasr111** [comment]
@felixdittrich92 Yes. We can close this ticket. Thanks for all the help :)
